lib/pure/times.nim

```diff
@@ -1210,15 +1373,32 @@ proc parse*(value, layout: string, zone: Timezone = local()): DateTime =
         token.add(layout[i])
         inc(i)
       else:
-        parseToken(dt, token, value, j)
+        found_tokens.incl parseToken(result, token, value, j)
         token = ""
 
-  if dt.isDst:
+  if parsedDateComplete * found_tokens != parsedDateComplete:
+      # Assumes current day of month, month and year
+    let dt = date()
+    result.monthday = dt.monthday
+    result.month = dt.month
+    result.year = dt.year
+
+  if ptTimeZone notin found_tokens:
     # No timezone parsed - assume timezone is `zone`
-    result = initDateTime(zone.zoneInfoFromTz(dt.toAdjTime), zone)
+    result = initDateTime(zone.zoneInfoFromTz(result.toAdjTime), zone)
   else:
     # Otherwise convert to `zone`
-    result = dt.toTime.inZone(zone)
+    result = result.toTime.inZone(zone)
+
+proc date*(s: string): Date = 
+  ## Parse `s` date string using ISO format
+  runnableExamples:
+    date"2019-03-21"
```

Comment:
Example should verify the result of `date"2019-03-21"`, this will likely complain about missing `discard`.

Reply:
I'm not really sure about this proc. If the use case is initialization, then IMO it should limit the argument to string literals so that `parseDate(str, "yyyy-MM-dd")` must be used when parsing input. A proc named `date` that does parsing seems a bit unclear.

Reply:
Removed this proc as requested. I can add `date` alias locally in my project.
I understand that it is deviation from the existing API,.
However, it is super convenient to have a single function `date`  that does all ways of date creation.

```
date"2019-02-03"
date(2019, mFeb, 3)
date(2019, 02, 03)
date() current date
date(x), where x can be EpochDate, Time, DateTime
```

So all you need to remember is the word `date` and you will be just fine with stdlib.
You can't do that effectively with `DateTime` as it has far too many members to keep specifying them one by one.

Please vote if you want `date` function back